.htaccess

```diff
@@ -1,5 +1,8 @@
-Header set Access-Control-Allow-Origin "*"
+SetEnvIf Origin "^http(s)?://.+$" AccessControlAllowOrigin=$0
```

Comment:
By default, che-devfile-registry doesn't have any sensitive information so I think it's ok.
But if we need to consider forked image, we need to have a configuration to make this strict.

Reply:
Sorry, I did not get why then just don't do it public, as it was previously.

Reply:
should not this value be set by OpenShift OAuth Proxy, if it's powered by it?

Reply:
Ah, what you meant is I should expose che-devfile-registry service as public, right?

In our case, we have two registries.
One is the che-devfile-registry and other is private registry.
che-devfile-registry is ok to be public since it's just quay.io/eclipse/che-devfile-registry.
But other is our own image and it has some credentials for development so it shouldn't be.
That's why I send this patch https://github.com/eclipse-che/che-dashboard/pull/333
But with `withCredentials` option, browser can't accept response if Access-Control-Allow-Credentials is not true and Access-Control-Allow-Origin is not the one even if a registry doesn't need credential.
So if che-dashboard can recognize a registry need credential or not, this change is not needed.

Reply:
> Ah, what you meant is I should expose che-devfile-registry service as public, right?

right.
it's a bit strange for a service that is supposed to be publicly available take care of authentication.

Ideal solution, I think is: Registry should be explicitly configured as private and then dashboard should know how to deal with it.

Reply:
> Registry should be explicitly configured as private and then dashboard should know how to deal with it.

I agree.
And now I found GitHub pages doesn't return Access-Control-Allow-Credentials header.
eclipse-che/che-dashboard#333 will throw an error if users are using GitHub pages for their registry.

Can you tell me the way to config? I'll implement that.
Adding CHE_WORKSPACE_DEVFILE__REGISTRY__WITH_CREDENTIAL_URL which accepts space separated urls?

Reply:
CheCluster already has devfile registry as an object.
And Che starting from 7.37 is going to have dashboard backend that potentially can directly read CheCluster or mounts registries as configmap...
let me think about it a bit more and return back with a proposal and you'll be able to implement.

Reply:
Should we revert this current pull request ?

Reply:
https://github.com/eclipse-che/che-plugin-registry/pull/1048
https://github.com/eclipse-che/che-devfile-registry/pull/467

Reply:
Issue to follows up the topic: Support for private devfile registries https://github.com/eclipse/che/issues/20465